exporters/otlp/src/otlp_http_metric_exporter.cc

```diff
@@ -0,0 +1,128 @@
+                                                            options.console_debug,
+                                                            options.timeout,
+                                                            options.http_headers
+#  ifdef ENABLE_ASYNC_EXPORT
+                                                            ,
+                                                            options.max_concurrent_requests,
+                                                            options.max_requests_per_connection
+#  endif
+                                                            )))
+{}
+
+OtlpHttpMetricExporter::OtlpHttpMetricExporter(std::unique_ptr<OtlpHttpClient> http_client)
+    : options_(OtlpHttpMetricExporterOptions()), http_client_(std::move(http_client))
+{
+  OtlpHttpMetricExporterOptions &options = const_cast<OtlpHttpMetricExporterOptions &>(options_);
+  options.url                            = http_client_->GetOptions().url;
+  options.content_type                   = http_client_->GetOptions().content_type;
+  options.json_bytes_mapping             = http_client_->GetOptions().json_bytes_mapping;
+  options.use_json_name                  = http_client_->GetOptions().use_json_name;
+  options.console_debug                  = http_client_->GetOptions().console_debug;
+  options.timeout                        = http_client_->GetOptions().timeout;
+  options.http_headers                   = http_client_->GetOptions().http_headers;
+#  ifdef ENABLE_ASYNC_EXPORT
+  options.max_concurrent_requests     = http_client_->GetOptions().max_concurrent_requests;
+  options.max_requests_per_connection = http_client_->GetOptions().max_requests_per_connection;
+#  endif
+}
+// ----------------------------- Exporter methods ------------------------------
+
+opentelemetry::sdk::common::ExportResult OtlpHttpMetricExporter::Export(
+    const opentelemetry::sdk::metrics::ResourceMetrics &data) noexcept
+{
+  if (http_client_->IsShutdown())
+  {
+    std::size_t metric_count = data.instrumentation_info_metric_data_.size();
+    OTEL_INTERNAL_LOG_ERROR("[OTLP HTTP Client] ERROR: Export "
+                            << metric_count << " metric(s) failed, exporter is shutdown");
+    return opentelemetry::sdk::common::ExportResult::kFailure;
+  }
+
+  if (data.instrumentation_info_metric_data_.empty())
+  {
+    return opentelemetry::sdk::common::ExportResult::kSuccess;
+  }
+  proto::collector::metrics::v1::ExportMetricsServiceRequest service_request;
+  OtlpMetricUtils::PopulateRequest(data, &service_request);
+  std::size_t metric_count = data.instrumentation_info_metric_data_.size();
+#  ifdef ENABLE_ASYNC_EXPORT
+  http_client_->Export(service_request, [metric_count](
+                                            opentelemetry::sdk::common::ExportResult result) {
+    if (result != opentelemetry::sdk::common::ExportResult::kSuccess)
+    {
+      OTEL_INTERNAL_LOG_ERROR("[OTLP HTTP Client] ERROR: Export "
```

Comment:
Return false when the call to `http_client_->Export` failed? Or what is the return value of the lambda for?

Reply:
It means this callback has already received and consumed this result, just like `Export` in OTLP HTTP exporters.

Reply:
> Return false when the call to `http_client_->Export` failed? Or what is the return value of the lambda for?

It means this callback has already received and consumed this result, just like Export in OTLP HTTP exporters.